File: ios/YearlyTracker/WidgetBridge.m
#import <React/RCTBridgeModule.h>

// Expose Swift module to React Native
@interface RCT_EXTERN_MODULE(WidgetBridge, NSObject)

RCT_EXTERN_METHOD(setWidgetPayload:(NSString *)payloadJson)
RCT_EXTERN_METHOD(clearWidgetPayload)
RCT_EXTERN_METHOD(setWidgetArtwork:(NSString *)sourceUri artworkId:(NSString *)artworkId)
RCT_EXTERN_METHOD(clearWidgetArtwork)

@end

File: ios/YearlyTracker/WidgetBridge.swift
import Foundation

@objc(WidgetBridge)
class WidgetBridge: NSObject {

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc(setWidgetPayload:)
  func setWidgetPayload(_ payloadJson: NSString) {
    SharedStore.savePayloadJSON(payloadJson as String)
  }

  @objc
  func clearWidgetPayload() {
    SharedStore.clearPayload()
  }

  @objc(setWidgetArtwork:artworkId:)
  func setWidgetArtwork(_ sourceUri: NSString, artworkId: NSString) {
    SharedStore.writeArtwork(from: sourceUri as String, artworkId: artworkId as String)
  }

  @objc
  func clearWidgetArtwork() {
    SharedStore.clearArtwork()
  }
}
